### ISSUE TITLE
syntax highlighting for Sublime may not be working in version 2.1

### ISSUE THREAD

**Issue by wilsonpilon** [farique1/basic-dignified#1]
Hello Rique. 

First of all I want to say that Basic Dignified is a great addition to MSX-BASIC development.

Until a few days ago I was using version 2.0 with Sublime and the syntax highlighting function was working Ok.

But this week I updated to version 2.1 in a clean installation (both Sublime and Basic Diginfied). Everything seems to work correctly (build, theme), except the syntax highlighting (and the snippets too)

It's not recognizing the extensions and also does not appear the MSX Basic option in "View"->"Syntax"

Here some details:

python --version
Python 3.11.2

subl --version
Sublime Text Build 4143

cd $env:APPDATA
cd '.\Sublime Text\Packages\User\'
dir

    Directory: C:\Users\wilso\AppData\Roaming\Sublime Text\Packages\User

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
da---          04/04/2023    10:32                BasicDignifiedMSX
-a---          30/03/2023    12:39            171 MSX Basic classic.sublime-settings
-a---          06/04/2023    09:01             42 MSX Basic Dignified.sublime-settings
-a---          04/04/2023    10:32             84 Package Control.sublime-settings
-a---          06/04/2023    10:17            125 Preferences.sublime-settings

cd .\BasicDignifiedMSX\build\
grep dropbox *

MSX Basic Dignified.sublime-build:              "cmd": ["python", "-u", "c:/dropbox/BasicDignified/badig.py",
MSX Basic Dignified.sublime-build:                              "cmd": ["python", "-u", "c:/dropbox/BasicDignified/badig.py",
MSX Basic Dignified.sublime-build:                              "cmd": ["python", "-u", "c:/dropbox/BasicDignified/badig.py",
MSX Basic Dignified.sublime-build:                              "cmd": ["python", "-u", "c:/dropbox/BasicDignified/badig.py",
MSX Basic classic.sublime-build:                "cmd": ["python", "-u", "c:/dropbox/BasicDignified/badig.py",
MSX Basic classic.sublime-build:                                "cmd": ["python", "-u", "c:/dropbox/BasicDignified/badig.py",
MSX Basic classic.sublime-build:                                "cmd": ["python", "-u", "c:/dropbox/BasicDignified/badig.py",
MSX Basic classic.sublime-build:                                "cmd": ["python", "-u", "c:/dropbox/BasicDignified/badig.py",
MSX Basic classic.sublime-build:                                "cmd": ["python", "-u", "c:/dropbox/BasicDignified/badig.py",

dir c:/dropbox/BasicDignified/badig.py

    Directory: C:\dropbox\BasicDignified

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---          30/03/2023    12:39          69426 badig.py

type '.\MSX Basic Dignified.sublime-settings'
{
        "extensions":
        [
                "bad",
                "dmx",
        ],
}

Thanks

**Comment by farique1** [comment]
Hi,
I'm on vacation right now. I'll take a look at that on Monday and report back.

**Comment by wilsonpilon** [comment]
Hello, don't worry. Enjoy your vacation!
cheers!

**Comment by wilsonpilon** [comment]
Rique, sorry for the alleged bug. Actually now I redid a totally clean installation (windows included) and the syntax highlighting reappeared (in syntax->user). Still does not recognize the extensions automatically (bad, dmx, etc..), and the snippets also do not load, I will investigate more.

**Comment by wilsonpilon** [comment]
Hello, don't worry. Enjoy your vacation!
cheers!

Rique, sorry for the alleged bug. Actually now I redid a totally clean installation (windows included) and the syntax highlighting reappeared (in syntax->user). The snippets also seem to work, so far the only thing that still doesn't seem to be the automatic extension recognization.

**Comment by farique1** [comment]
Thanks!
Back to the real life :)

I think I may have found the issue.
I changed the structure of the package (moving the files to subfolders) on v2.0 (or was it v2.1?) and it may have confused Sublime.
Also I replaced spaces with - on the language scopes to standardize with the VSCode extension. I think Sublime uses the name of the .sublime-settings files to identify the correct scope. Try reaming both .sublime-settings replacing the spaces with - and tell me if it works.

By the way, thanks for your MSX Basic extension on VSCode, it helped a lot when building my own extension!
If you have tips, comments and features requests about Basic Dignified, fell free to get in touch.
As an example, I noticed that you INCLUDEd a file with only DEFINEs in it. This no longer works on 2.0 as all identifiers are exclusive to their own files. I was unsure when I took that decision even before I saw how you were using it. What do you think about the sharing of the identifiers between INCLUDEd files (labels, defines, etc)?

Cheers!

**Comment by wilsonpilon** [comment]
Welcome back!

The structure changes occours from 2.0 to 2.1.

Following your suggestion to rename from " " to "-", I additionally had to change the filenames to lower case, looking like this:

msx-basic-classic.sublime-settings
msx-basic-dignified.sublime-settings

This way it worked correctly

Wow, no need to thank me, my extension was just a little toy, but if it helped in any way I'm glad. I even need to finish a more updated version of it.

As for the defines in separate files, there are two approaches, I use them separately to have some ready made kits with definitions for specific types of programs and another with definitions for MSX itself, it's more an organizational thing from my heritage with C, Unix, LIBs and computers with few resources. I particularly prefer to work with multiple smaller files that add similar functions and definitions and that can be reused.

However, unique definition and basic file in more modern environments with more powerful editors that allow multiple frames of the same file ends up being cool too. The approach you follow is fine.

**Comment by farique1** [comment]
Cool!
I'll update the files and reupload the project with the changes so far.
Thanks for the reports.

Actually I've come to see Basic Dignified more and more as a compiler. If you look at the early versions there is a ton of 
aesthetic customization options for the output code. I removed most of them as I began to see the classic output as a compiled thing that you rarely have a reason to mess with.
I'll make a note on the "maybe todo" section to offer an option to turn on and off the independence of the INCLUDEs.
